scala-package/core/src/main/scala/org/apache/mxnet/Model.scala

```diff
@@ -290,66 +290,69 @@ object Model {
       var epochDone = false
       // Iterate over training data.
       trainData.reset()
-      while (!epochDone) {
-        var doReset = true
-        while (doReset && trainData.hasNext) {
-          val dataBatch = trainData.next()
-          executorManager.loadDataBatch(dataBatch)
-          monitor.foreach(_.tic())
-          executorManager.forward(isTrain = true)
-          executorManager.backward()
-          if (updateOnKVStore) {
-            updateParamsOnKVStore(executorManager.paramArrays,
-              executorManager.gradArrays,
-              kvStore, executorManager.paramNames)
-          } else {
-            updateParams(executorManager.paramArrays,
-              executorManager.gradArrays,
-              updaterLocal, ctx.length,
-              executorManager.paramNames,
-              kvStore)
-          }
-          monitor.foreach(_.tocPrint())
-          // evaluate at end, so out_cpu_array can lazy copy
-          executorManager.updateMetric(evalMetric, dataBatch.label)
+//      ResourceScope.using() {
```

Comment:
Yes, that is because executorManager.gradArrays/ executorManager.paramArrays all of these are initialized on the fly during initialization of trainExecs, they will deAllocated at the end of scope so  it crashes in the next epoch.

Reply:
I figure out the problem, its in the Optimizer createState. I'll find a way workaround it.